src/tools/compiletest/src/runtest.rs

```diff
@@ -1951,6 +1951,9 @@ impl<'test> TestCx<'test> {
                 if !self.props.compile_flags.iter().any(|s| s.starts_with("--error-format")) {
                     rustc.args(&["--error-format", "json"]);
                 }
+                if !self.props.compile_flags.iter().any(|s| s.contains("codegen-units")) {
```

Comment:
This condition should be unnecessary because options passed later override options passed earlier, and options from `// compile-flags` are passed later to give individual tests chance to override defaults.

For example, `-Zdeduplicate-diagnostics=no` below is passed unconditionally, but some tests override it with `// compile-flags: -Zdeduplicate-diagnostics=yes`.